packages/graphql/src/translate/queryAST/cypher-generators/CypherAnnotationSubqueryGenerator.ts

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Cypher from "@neo4j/cypher-builder";
+import { QueryASTContext } from "../ast/QueryASTContext";
+import type { AttributeAdapter } from "../../../schema-model/attribute/model-adapters/AttributeAdapter";
+import type { CypherAnnotation } from "../../../schema-model/annotation/CypherAnnotation";
+import type { Field } from "../ast/fields/Field";
+import type { CypherUnionAttributePartial } from "../ast/fields/attribute-fields/CypherUnionAttributePartial";
+import { assertIsCypherNode } from "../utils/is-cypher-node";
+import { wrapSubqueryInCall } from "../utils/wrap-subquery-in-call";
+
+/** Variable exposed to the user in their custom cypher */
+const CYPHER_TARGET_VARIABLE = new Cypher.NamedVariable("this");
+
+export class CypherAnnotationSubqueryGenerator {
```

Comment:
I like the approach of refactoring these functions using a class, going further, I would separate the different behavior "Union" vs "Concrete" in two different concrete classes with both implementing the public method "createSubqueryForCypherAnnotation"!